### ISSUE TITLE
Dynamic Group - UI Issues with List View

### ISSUE THREAD

**Issue by mzbroch** [nautobot/nautobot#2140]
### Environment
* Python version:
* Nautobot version: 1.4.0-beta1 (branch next from 4th of August)

### Steps to Reproduce
1. Create following dynamic groups:
- Cisco
- Juniper
- Leaf
- Spine
- Salt-Service

2. Stack them as below:
```
Salt-service
| Cisco
| - Leaf
| - Spine
| Juniper
| - Leaf
| - Spine
```
so that the top service (salt) has both Cisco and Juniper. Each Cisco and Juniper has Leafs and Spines

### Expected Behavior

Sorted as per the information above.

### Observed Behavior

Sorting as on the screenshot:
![image](https://user-images.githubusercontent.com/16935122/182855174-7fe57a92-882d-48b2-a1ed-d1e2cb778dc3.png)

**Comment by gsnider2195** [comment]
It looks like this is sorting by weight and not taking into account inheritance

**Comment by jathanism** [comment]
I think it's a basic bug in the UI code that is mapping objects based on their slug and a weight, and if there are duplicates the most recent one is being included incorrectly. That's just a hunch though.

**Comment by bryanculver** [comment]
Closed with #2144
